_example/cmd/database/others.go

```diff
@@ -46,13 +46,42 @@ func (db *Database) ListAvailableIdentityProviders() []resources.AvailableIdenti
 	return db.AvailableIdentityProviders
 }
 
-// ListUserResources returns the list of resources.
-func (db *Database) ListUserResources() []resources.Resource {
+// ListUserResources returns the list of resources. If entityType is nil, the
+// method returns all resources.
+func (db *Database) ListUserResources(entityType *string) []resources.Resource {
 	db.mutex.RLock()
 	defer db.mutex.RUnlock()
 	db.Load()
 
-	return db.Resources
+	// Create a map of resources by flattening the hierarchical data structure.
+	leaves := append([]resources.Resource{}, db.Resources...)
+	m := map[resources.Resource]resources.Resource{}
+	for {
+		parents := []resources.Resource{}
+		for _, l := range leaves {
+			if l.Parent != nil {
+				parents = append(parents, *l.Parent)
+			}
+
+			key := l
+			key.Parent = nil
+			if _, ok := m[key]; !ok {
```

Comment:
question: is it "safe"/"ok" to use the whole object as a hashable key for the map?

Reply:
Yeah, it works with Go maps. The only thing is memory usage which is okay with this in-memory example implementation.